QuickFIXn/Session.cs

```diff
@@ -362,7 +363,19 @@ public bool Send(string message)
             {
                 if (_responder is null)
                     return false;
-                Log.OnOutgoing(message);
+
+                const LogLevel messagesLogLevel = LogLevel.Information;
+                if (Log.IsEnabled(messagesLogLevel))
+                {
+                    using (Log.BeginScope(new Dictionary<string, object>
```

Comment:
@jkulubya  What is this scope doing?  My brain is refusing to understand what this code does.

Reply:
BeginScope adds the parameters from the dictionary for all logging done inside that using scope

So in serilog land for json sink you would have what ever props are in the dictionary with the log so in this case 
Properties would be this for a fix44 AllocationReport message

{
  "Message": "The actual message..."
  "Properties" : {"MessageType": "AllocationReport"}
}


Azure App insights would be similar though would end up in custom dimensions I think its called